docs/design/proposal-troubleshoot-consolidated-cli.md

```diff
@@ -0,0 +1,243 @@
+
+## Non-Goals
+
+## Background
+
+## High Level Design
+
+Functions of `support-bundle`, `preflight`, `analyze`, `redact`, and `sbctl` binaries/tools should be rolled together into a single `troubleshoot` CLI that can perform all necessary functions of the same.
+
+`troubleshoot` CLI should be able to report on the version that is installed in the CLI, and any support bundle generated with `troubleshoot` should report its build/version inside the archive it generates.
+
+## Detailed Design
+
+In the interest of being able to work on this quickly without breaking existing use-cases, a new `troubleshoot` command should be created. Utilizing cobra and viper best practices from the cobra.dev docs.
+
+### sbctl → troubleshoot inspect
+
+sbctl should be migrated to the troubleshoot repository in a "lift and shift" operation to start with.
+
+It should continue to be built as a standalone binary to enable continued use until a stable replacement exists. as such it should me integrated in a way that preserves the existing `cmd` and it's package requirements.
+
+Once integrated into the repository, we can begin using functions from it's packages from the from the new `troubleshoot` command to enable the `inspect` subcommand.
+
+### Public APIs
+
+The existing package entrypoint we use from the support-bundle command is `supportbundle.CollectSupportBundleFromSpec()`
+which is an end-to-end function that runs collectors, redactors and then analysis.
+
+The Consolidated cli should be responsible for handling the end-to-end running of this process in the `cmd` leaving the importable packages free to focus on defining the individual collect, redact analyze steps.
+
+To this end we should create a new `pkg` that targets the functionality provided by the `collect` and `analyze` packages such as `collect.runHostCollectors()` and provide a stable API to be used by the CLI and other projects. At the same time we can unpublish functionality that should not be exposed (such as the individual collector and analyser functions) and mark code for deprecation.
+
+This new package should be kept as minimal as possible. and serve only as an interface to private functions in the other packages.
+
+Once the stable API is ready we can instruct projects like kurl to target that and work on removing code marked for deprecation.
+
+The functionality we want to expose via this api is:
+
+- `CollectBundle(spec{}) (bundlePath string, error)`
+  - collect a support bundle from a spec.
+  - takes a parsed spec struct as an parameter.
+  - returns a path to the bundle directory and errors.
+  - to minimise IO and increase collection speed. redactors should be run inline, redacting data in memory before it's saved to a bundle.
+- `RedactBundle(bundlePath string) error`
+  - redact an already collected bundle, takes a path to the bundle as an parameter.
+  - returns any errors
```

Comment:
Could this API accept list of redactors with non-hardcoded defaults. I think that currently there is no way to disable some default collectors, like IP addresses.

Reply:
IP addresses was removed from the defaults, but the others could use more granular switches, for sure.